azure-pipelines.yml

```diff
@@ -0,0 +1,19 @@
+# https://docs.microsoft.com/azure/devops/pipelines/languages/android
+jobs:
+  - job: gradle_junit_tests
+    steps:
+      - task: Gradle@2
+        inputs:
+          workingDirectory: espresso-server
+          gradleWrapperFile: espresso-server/gradlew
+          gradleOptions: '-Xmx3072m'
+          publishJUnitResults: true
+          testResultsFiles: '**/TEST-*.xml'
+          tasks: test
+  - template: azure-pipelines/android-e2e-template.yml
+    parameters:
+      script: npx mocha --timeout 6000000 --reporter mocha-junit-reporter --recursive build/test/functional/ -g @skip-ci -i --exit
+      name: sdk28_e2e_tests
```

Comment:
sdk${ANDROID_SDK_VERSION}_e2e_tests ?

Reply:
No, the YML can't have environment variables in it.... templates can have parameter variables.